### PR TITLE
[FEATURE] Customizable "Suite Edit" generated notebooks

### PR DESCRIPTION
**Pull request by dmateusp**
Changes proposed in this pull request:
- Refactor great_expectations/render/renderer/suite_edit_notebook_renderer.py to bring different sections from .py or .md files
- Add a configuration section to overwrite suite_edit generated notebooks
- Make all markdown / code template-able through Jinja2

Previous Design Review notes:
- https://github.com/great-expectations/great_expectations/issues/1643

### CODE REVIEW

File: docs/how_to_guides/configuring_generated_notebooks/how_to_configure_suite_edit_generated_notebooks.rst [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+Each one of those cells expects either Markdown or code. Great Expectations has defaults for each
+one of those sections, but it supports each section to be overwritten. The default class supports
+using Jinja templating for dynamic rendering.
+
+Additionally, the class that generates the notebooks can be overwritten.
+
+
+----------------------
+Overwritting the class
+----------------------
```
Comment (Aylr) [inline docs/how_to_guides/configuring_generated_notebooks/how_to_configure_suite_edit_generated_notebooks.rst]:
```suggestion
---------------------
Overwriting the class
---------------------
```

File: docs/how_to_guides/configuring_generated_notebooks/how_to_configure_suite_edit_generated_notebooks.rst [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+
+        notebooks:
+            suite_edit:
+                class_name: CustomeSuiteEditNotebookRender
+                module_name: custom_plugin.render.renderer.suite_edit_notebook_renderer
+
+
+----------------------
+Overwritting a section
+----------------------
```
Comment (Aylr) [inline docs/how_to_guides/configuring_generated_notebooks/how_to_configure_suite_edit_generated_notebooks.rst]:
```suggestion
----------------------
Overwriting a section
----------------------
```

**Review by Aylr — CHANGES_REQUESTED** [review]
This is phenomenal work @dmateusp and really makes these notebooks more flexible. I particularly love the detail put into the howto guide! I am also delighted that you have done this without breaking the existing API. Bravo!

My only change request are a few typos, and adding a line to the changelog about this awesome new feature.

**Comment by Aylr** [comment]
Also, my apologies for taking so long to review this!

**Comment by dmateusp** [comment]
Thanks for the feedback @Aylr ! I commit-ed your suggestions and updated the changelog

**Pull request merged**